### PR TITLE
Revert "rustc: build with jemalloc"

### PR DESCRIPTION
**Pull request by zowoq**
This reverts commit 2a699029f43a2d467b07fc42bea8fd975ffa766f.

hopefully this fixes building x86_64-darwin under emulation on aarch64-darwin

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by winterqt** [comment]
What about the linked issue (https://github.com/rust-lang/rust/issues/92173), though...? Can we risk that?

**Comment by zowoq** [comment]
> Can we risk that?

I'd rather not but hopefully this will unblock staging-next and buy us a bit of time.

**Comment by winterqt** [comment]
I have a more proper solution for this, but don't have any more time today to run what I hope to be the final test build of `rustc`/`cargo`/`fd` (takes a while under Rosetta).

If I can get this working (tomorrow), I'll PR it. If not, we can merge this to unblock so we have more time to figure this out.

Unless you want to merge this now instead of waiting -- let me know.

**Comment by zowoq** [comment]
> I have a more proper solution for this

Can you push it to a branch? Maybe we can get someone else to test it?

**Comment by winterqt** [comment]
https://github.com/winterqt/nixpkgs/commit/b34decf9adadd8a38337ba4271ac8f84416c04d0

**Comment by winterqt** [comment]
After actually reading the comment I was basing this off of (https://github.com/rust-lang/rust/issues/92173#issuecomment-1084957372), it looks like my method (building with the 11.0 SDK) probably won't fix this.

We could always give it a try (https://github.com/winterqt/nixpkgs/commit/b34decf9adadd8a38337ba4271ac8f84416c04d0), but sadly it looks like we either have to fix compiling jemalloc under Rosetta (I may take a stab at this, though I have no idea what I'm doing), or keep a native x86_64-darwin builder around. I say these are the only options because it looks like the SDK bump wouldn't even help here, as it wouldn't get us to Xcode 13.3. (I have no clue what SDK version these issues started at though, so...)

**Comment by zowoq** [comment]
> https://github.com/winterqt/nixpkgs/commit/b34decf9adadd8a38337ba4271ac8f84416c04d0

I'd be a bit concerned about having `rustc` on 11.0 and `cargo`/`buildRustpackage` on 10.12, not sure if it might need other changes as well? https://github.com/NixOS/nixpkgs/pull/180931 is also working on moving rust to 11.0.

**Comment by winterqt** [comment]
> https://github.com/NixOS/nixpkgs/pull/180931 is also working on moving rust to 11.0.

Yup, I know -- this was just a stopgap (if it actually fixed the issue at hand), and might've caused some issues for a few packages.

(Though Go hasn't had any issues with this, AFAIK?)

**Comment by zowoq** [comment]
> Though Go hasn't had any issues with this, AFAIK?

Don't think there has been any issues but may not be a useful indicator, relatively few go packages use sdk features that aren't already propagated by go (and it only uses a couple).

At the moment I think merging this PR is probably the least bad option, for the next staging cycle we can try switching to the newer sdk or finding a jemalloc fix.

**Comment by winterqt** [comment]
> At the moment I think merging this PR is probably the least bad option, for the next staging cycle we can try switching to the newer sdk or finding a jemalloc fix.

Agreed -- let's buy us some time, even if this may be a ticking time bomb (maybe we should reopen #144704 so people are aware that this probably will be occurring on unstable?)

**Review by winterqt — APPROVED** [review]

**Comment by zowoq** [comment]
> maybe we should reopen https://github.com/NixOS/nixpkgs/issues/144704 so people are aware that this probably will be occurring on unstable?

Good idea but I'll wait a bit until we're closer to it landing in master.

**Comment by zowoq** [comment]
Reopened https://github.com/NixOS/nixpkgs/issues/144704.

**Pull request merged**
